enchant-2.2.15 make check failure:enchant_dict_add: assertion 'dict' failed

When make check on enchant@2.2.15,the following error occurs:
==========================================
   enchant 2.2.15: tests/test-suite.log
==========================================

# TOTAL: 1
# PASS:  0
# SKIP:  0
# XFAIL: 0
# FAIL:  1
# XPASS: 0
# ERROR: 0

.. contents:: :depth: 2

FAIL: main
==========
** (process:91891): CRITICAL **: 16:28:44.009: enchant_dict_add: assertion 'dict' failed

** (process:91891): CRITICAL **: 16:28:44.009: enchant_dict_check: assertion 'dict' failed
dictionary/enchant_dict_add_tests.cpp:87:1: error: Failure in EnchantDictionaryAdd_WordExistsInDictionary: IsWordInDictionary("hello")

** (process:91891): CRITICAL **: 16:28:44.010: enchant_dict_add: assertion 'dict' failed

** (process:91891): CRITICAL **: 16:28:44.010: enchant_dict_is_added: assertion 'dict' failed
dictionary/enchant_dict_add_tests.cpp:94:1: error: Failure in EnchantDictionaryAdd_WordExistsInSession: IsWordInSession("hello")

** (process:91891): CRITICAL **: 16:28:44.012: enchant_dict_add: assertion 'dict' failed
dictionary/enchant_dict_add_tests.cpp:102:1: error: Failure in EnchantDictionaryAdd_WordAddedToEnchantPwlFile: PersonalWordListFileHasContents()

** (process:91891): CRITICAL **: 16:28:44.013: enchant_dict_add: assertion 'dict' failed
dictionary/enchant_dict_add_tests.cpp:109:1: error: Failure in EnchantDictionaryAdd_PassedOnToProvider_LenComputed: addToPersonalCalled
dictionary/enchant_dict_add_tests.cpp:110:1: error: Failure in EnchantDictionaryAdd_PassedOnToProvider_LenComputed: Expected hello but was

** (process:91891): CRITICAL **: 16:28:44.015: enchant_dict_add: assertion 'dict' failed
dictionary/enchant_dict_add_tests.cpp:117:1: error: Failure in EnchantDictionaryAdd_PassedOnToProvider_LenSpecified: addToPersonalCalled
dictionary/enchant_dict_add_tests.cpp:118:1: error: Failure in EnchantDictionaryAdd_PassedOnToProvider_LenSpecified: Expected hello but was

** (process:91891): CRITICAL **: 16:28:44.017: enchant_dict_add_to_session: assertion 'dict' failed

** (process:91891): CRITICAL **: 16:28:44.017: enchant_dict_add: assertion 'dict' failed
dictionary/enchant_dict_add_tests.cpp:129:1: error: Failure in EnchantDictionaryAdd_WordExistsInSession_StillCallsProvider: addToPersonalCalled
dictionary/enchant_dict_add_tests.cpp:130:1: error: Failure in EnchantDictionaryAdd_WordExistsInSession_StillCallsProvider: Expected session but was

** (process:91891): CRITICAL **: 16:28:44.018: enchant_dict_add: assertion 'dict' failed

** (process:91891): CRITICAL **: 16:28:44.018: enchant_dict_add: assertion 'dict' failed
dictionary/enchant_dict_add_tests.cpp:141:1: error: Failure in EnchantDictionaryAdd_WordExistsInPersonal_StillCallsProvider: addToPersonalCalled
dictionary/enchant_dict_add_tests.cpp:142:1: error: Failure in EnchantDictionaryAdd_WordExistsInPersonal_StillCallsProvider: Expected personal but was

** (process:91891): CRITICAL **: 16:28:44.020: enchant_dict_remove: assertion 'dict' failed
dictionary/enchant_dict_add_tests.cpp:149:1: error: Failure in EnchantDictionaryAdd_WordExistsInExclude_RemovedFromExcludeAddedToPersonal: ExcludeFileHasContents()

** (process:91891): CRITICAL **: 16:28:44.020: enchant_dict_add: assertion 'dict' failed

--- a/NEWS
+++ b/NEWS
@@ -1,3 +1,10 @@
+2.6.3 (December 2, 2023)
+-----------------------
+
+This version fixes a bug in the tests when pkgdatadir is set to a
+non-default value, and clarifies the documentation for setting pkgdatadir.
+
+
 2.6.2 (October 28, 2023)
 ------------------------
 

--- a/configure.ac
+++ b/configure.ac
@@ -13,7 +13,7 @@
 # along with this program; if not, see <https://www.gnu.org/licenses/>.
 
 AC_PREREQ([2.71])
-AC_INIT([enchant],[2.6.2])
+AC_INIT([enchant],[2.6.3])
 AC_CONFIG_SRCDIR(src/enchant.h)
 AC_CONFIG_AUX_DIR([build-aux])
 AM_INIT_AUTOMAKE([subdir-objects])

--- a/gl/doc/INSTALL.diff
+++ b/gl/doc/INSTALL.diff
@@ -27,7 +27,7 @@
 +versions can be installed in parallel.  The one exception is the data files
 +(under pkgdatadir), as these are compatible between different major
 +versions.  To change e.g. /usr/share/enchant to /usr/share/enchant-2, set
-+pkgdatadir when running make, e.g.:
++pkgdatadir when running any make command (including check & install), e.g.:
 +
 +make pkgdatadir=/usr/share/enchant-2
 +

--- a/tests/EnchantTestFixture.h
+++ b/tests/EnchantTestFixture.h
@@ -38,6 +38,7 @@
 #include <codecvt>
 
 #include "enchant.h"
+#include "enchant-provider.h"
 
 struct EnchantTestFixture
 {
@@ -66,7 +67,9 @@ struct EnchantTestFixture
 
     std::string GetEnchantConfigDir()
     {
-        return AddToPath("share", "enchant");
+        GSList *config_dirs = enchant_get_conf_dirs();
+        const char *pkgdatadir = (char *)g_slist_nth(config_dirs, 0)->data;
+        return std::string(pkgdatadir);
     }
 
     static void DeleteDirAndFiles(const std::string& dir)

